Tests/XCUITests/JumpBackInTests.swift

```diff
@@ -0,0 +1,182 @@
+        navigator.nowAt(NewTabScreen)
+    }
+
+    func scrollDown() {
+        if !isTablet {
+            while app.staticTexts["Switch Your Default Browser"].exists || app.buttons["Learn How"].exists {
+                app.collectionViews["FxCollectionView"].swipeUp()
+            }
+        }
+    }
+
+    override func setUp() {
+        super.setUp()
+
+        closeKeyboard()
+
+        // "Jump Back In" is enabled by default. See Settings -> Homepage
+        navigator.goto(HomeSettings)
+        XCTAssertEqual(app.switches["Jump Back In"].value as! String, "1")
+
+        navigator.goto(NewTabScreen)
+    }
+
+    func testJumpBackInSection() {
+        // Open a tab and visit a page
+        navigator.openURL("https://www.example.com")
+        waitUntilPageLoad()
+
+        // Open a new tab
+        navigator.goto(TabTray)
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        closeKeyboard()
+
+        // "Jump Back In" section is displayed
+        waitForExistence(app.staticTexts["Jump Back In"])
+
+        // The contextual hint box is not displayed consistently, so
+        // I don't test for its existence.
+    }
+
+    func testGroupedTabs() {
+        // Enable "Tab Groups" from Settings -> Tabs
+        navigator.goto(TabsSettings)
+        navigator.performAction(Action.ToggleTabGroups)
+        navigator.goto(SettingsScreen)
+        app.navigationBars.buttons["Done"].tap()
+        navigator.nowAt(NewTabScreen)
+
+        // Create 3 groups in tab tray
+        let groups = ["test1", "test2", "test3"]
+        for group in groups {
+            for _ in 1...3 {
+                navigator.goto(TabTray)
+                navigator.performAction(Action.OpenNewTabFromTabTray)
+                navigator.openURL(group)
+                waitUntilPageLoad()
+            }
+        }
+
+        // Open a new tab
+        navigator.goto(TabTray)
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        closeKeyboard()
+
+        // Tap on the "test3" from "Jump Back In" section
+        scrollDown()
+        waitForExistence(app.staticTexts["Jump Back In"])
+        app.cells["JumpBackInCell"].staticTexts["Test3"].tap()
+        if isTablet {
+            waitForExistence(app.navigationBars.segmentedControls["navBarTabTray"])
+        } else {
+            waitForExistence(app.navigationBars.staticTexts["Open Tabs"])
+        }
+        waitForExistence(app.staticTexts["Test3"])
+    }
+
+    func testPrivateTab() {
+        // Visit https://www.twitter.com
+        navigator.openURL("https://www.twitter.com")
+        waitUntilPageLoad()
+
+        // Open a new tab and check the "Jump Back In" section
+        navigator.goto(TabTray)
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        closeKeyboard()
+
+        // Twitter tab is visible in the "Jump Back In" section
+        scrollDown()
+        waitForExistence(app.staticTexts["Jump Back In"])
```

Comment:
This test is failing on this line. What I think is the cause, also after looking that the screenshots.. is that we go to the tab tray but we don't wait to have it completely loaded. Not sure why it does not fail in creating the new tab, but I don't see it created and so we never go to the Homepanel view and so the test fails on this line.... maybe adding waitForElements in the tab tray and to the close button will help to debug this...

Reply:
> This test is failing on this line. What I think is the cause, also after looking that the screenshots.. is that we go to the tab tray but we don't wait to have it completely loaded. Not sure why it does not fail in creating the new tab, but I don't see it created and so we never go to the Homepanel view and so the test fails on this line.... maybe adding waitForElements in the tab tray and to the close button will help to debug this...

🤔 💭 I have taken a look in the Screenshots from Bitrise test reports. If the test actually failed to open a new tab on line 99 or 77, `closeKeyboard()` should be failing because the function `waitForExistence` on `app.buttons["urlBar-cancel"]`, which is on a new tab screen. At least some screenshots shows a screen that is still transitioning from the "Open Tabs" page.

On my local computer, I've tested both `testGroupedTabs()` and `testPrivateTabs()` using `-test-iterations 10` and both tests are passing consistently on my M1. 😕

Reply:
Same here... both works... it is a timing issue around that for sure.. I know it should fail trying to close the urlBar.. but all the screenshots I see are either transitioning to Tab Tray or on that view...